docker-compose.yml

```diff
@@ -21,32 +21,52 @@ slave:
         - jenkins
     volumes:
         - ./slave:/home/slave
+        - ./nginx/conf.d:/home/slave/nginx
     volumes_from:
         - git
+        - nginx
     environment:
         - JAVA_HOME=/etc/alternatives/java_sdk
         - SERVICE_NAME=SPACENAME-omero
         - DOCKER_FIX='   '
-    command: -labels java -labels slave -name slave -executors 4
+    command: -labels java -labels slave -name slave -disableClientsUniqueId -executors 4
 
 web:
     build: web
-    environment:
-        - JENKINS_LABEL=/etc/alternatives/java_sdk
-        - JENKINS_MODE=exclusive
-        - SERVICE_NAME=SPACENAME-omero
+    ports:
+        - "4080:4080"
+    privileged: true
     links:
         - jenkins
+        - redis
     volumes:
         - ./web:/home/omero
-    command: -labels web -name web -executors 4
+        - ./nginx/conf.d:/home/omero/nginx
+    volumes_from:
+        - nginx
+    environment:
+        - JENKINS_LABEL=/etc/alternatives/java_sdk
+        - JENKINS_MODE=exclusive
+        - SERVICE_NAME=SPACENAME-web
 
 nginx:
-    image: nginx
+    build: nginx
+    ports:
+        - "8888:80"
+    privileged: true
     links:
-        - slave
-        - web
+        - jenkins
     volumes:
-        - ./slave:/home/jenkins-slave
+        - ./nginx/conf.d:/etc/nginx/conf.d
+        - ./web/OMERO.server/lib/python/omeroweb/static:/home/omero/nginx/web/static
     environment:
-        - SERVICE_NAME=SPACENAME-web
+        - JENKINS_LABEL=/etc/alternatives/java_sdk
+        - JENKINS_MODE=exclusive
+        - SERVICE_NAME=SPACENAME-nginx
+
+redis:
+    image: redis
+    ports:
+        - "6379:6379"
```

Comment:
At this moment I am debating how to dynamically recognise redis host and port in Jenkins job. I was wondering if it would be good idea to introduce Jenkins node to be mounted a volume in another nodes. That would remove @manics hack as well, any thoughts?